bazaarvoice/emodb
# Fixed use of Ticker for getting current time

## Github Issue #

[39](https://github.com/bazaarvoice/emodb/issues/39)

## What Are We Doing Here?

There are several changes in this PR:

1. Replaced locations where `Ticker` was being used to get the current time with a `Clock`.
2. Changed `Clock` and `Ticker` instances to be formally injected by Guava.

## How to Test and Verify

1. Check out the project
2. Run EmoDB
3. Configure a Stash server with a scheduled scan and start Stash. The startup logs will show the first scan scheduled for the next expected time (within 24 hours).

## Risk

Low.  The new classes introduced are thin and mostly self explanatory; otherwise, most of the changes are around injection of the new time providers.

### Level 

`Medium`

### Required Testing

`Regression`

### Risk Summary

Low; associated risks are already outlined in the corresponding issue.

## Code Review Checklist


build and run.

we inevitably come back to this code it will only take hours to figure out, not
days.

a victim of rampaging consistency, and should be using this course of action. 
We don't have coding standards out yet for this project, so please make sure to address any feedback regarding STYLE so the codebase remains consistent.


…time
